coral-component-table/src/scripts/Table.js

```diff
@@ -2467,6 +2467,12 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
       headerCell.setAttribute('role', 'rowheader');
     }
     headerCell.setAttribute('scope', scope);
+    if(!headerCell.children.length > 0) {
+      headerCell._elements.content.setAttribute('role', 'button');
+      if(!headerCell.firstChild) {
+        headerCell._elements.content.setAttribute('role', 'columnheader');
+      }
```

Comment:
headercell will be a columnheader, why are setting role to button ??

Reply:
The ticket requires that the interactive elements of the headercell must have a button role 
![SITES-10962](https://user-images.githubusercontent.com/101817370/236141238-05aabbc4-85ff-44a5-9af4-8dc1528692f0.png)

Reply:
Hi @StoianLucian ,
The ticket suggestion is not always the best and feasible solution and might not be aware of the component usage.  The exact problem is the headercell doesn't have a role, you need to add the most apt role to it.

Reply:
Hi @Pareesh Since the interactive elements act as a sorting button for the content below (**SITES-10962.png**) won`t the role 'button" be the most appropriate one?
![SITES-10962](https://github.com/adobe/coral-spectrum/assets/101817370/0f24ddb1-a77e-4dd4-9c9d-20ff2762cf07)

Reply:
Hi @Pareesh I changed the logic to apply the button role only if the parent element (wrapper) has the sortable attribute. It doesn`t change the role of the non-interactive rows

Reply:
HI @Pareesh I had to move the functionality in order to apply the changes since the sortable attribute was applied later in the original function